tools/clang/lib/CodeGen/ModuleBuilder.cpp

```diff
@@ -235,14 +236,16 @@ namespace {
                  end = Ctx.getSourceManager().fileinfo_end();
              it != end; ++it) {
           if (it->first->isValid() && !it->second->IsSystemFile) {
+            std::string filename = it->first->getName();
+            std::replace(filename.begin(), filename.end(), '\\', '/');
             // If main file, write that to metadata first.
             // Add the rest to filesMap to sort by name.
             if (CodeGenOpts.MainFileName.compare(it->first->getName()) == 0) {
```

Comment:
If we can normalize paths earlier for SourceMap, we must also normalize the MainFileName path to match.